File: app/src/test/java/com/slapshotapps/dragonshockey/activities/admin/viewmodels/AdminGameViewModelTest.java
package com.slapshotapps.dragonshockey.activities.admin.viewmodels;

import com.slapshotapps.dragonshockey.models.Game;
import com.slapshotapps.dragonshockey.models.GameResult;
import junit.framework.Assert;
import org.junit.Test;

/**
 * Created on 10/23/16.
 */
public class AdminGameViewModelTest {
    @Test
    public void getOpponentName_null() {
        Game game = new Game();

        AdminGameViewModel adminGameViewModel = new AdminGameViewModel(game);

        Assert.assertNotNull(adminGameViewModel.getOpponentName());
    }

    @Test
    public void setOpponentName_null() {
        Game game = new Game();

        AdminGameViewModel adminGameViewModel = new AdminGameViewModel(game);
        adminGameViewModel.setOpponentName("test");

        Assert.assertEquals("test", adminGameViewModel.getOpponentName());

        Assert.assertTrue(adminGameViewModel.hasChanged());
    }

    @Test
    public void setOpponentName_notNull() {
        Game game = new Game();
        game.opponent = "beforeTest";

        AdminGameViewModel adminGameViewModel = new AdminGameViewModel(game);
        adminGameViewModel.setOpponentName("test");

        Assert.assertEquals("test", adminGameViewModel.getOpponentName());

        Assert.assertTrue(adminGameViewModel.hasChanged());
    }

    @Test
    public void setOpponentScore_null() {
        Game game = new Game();

        AdminGameViewModel adminGameViewModel = new AdminGameViewModel(game);
        adminGameViewModel.setOpponentScore(null);

        Assert.assertEquals("0", adminGameViewModel.getOpponentScore());
        Assert.assertTrue(adminGameViewModel.hasChanged());
    }

    @Test
    public void setOpponentScore_nullGameResult() {
        Game game = new Game();

        AdminGameViewModel adminGameViewModel = new AdminGameViewModel(game);
        adminGameViewModel.setOpponentScore("8");

        Assert.assertEquals("8", adminGameViewModel.getOpponentScore());
        Assert.assertTrue(adminGameViewModel.hasChanged());
    }

    @Test
    public void setOpponentScore_notNull() {
        Game game = new Game();
        game.gameResult = new GameResult();
        game.gameResult.opponentScore = 2;

        AdminGameViewModel adminGameViewModel = new AdminGameViewModel(game);
        adminGameViewModel.setOpponentScore("8");

        Assert.assertEquals("8", adminGameViewModel.getOpponentScore());
        Assert.assertTrue(adminGameViewModel.hasChanged());
    }

    @Test
    public void getOpponentScore() throws Exception {
        Game game = new Game();

        AdminGameViewModel adminGameViewModel = new AdminGameViewModel(game);

        Assert.assertEquals("", adminGameViewModel.getOpponentScore());
        Assert.assertFalse(adminGameViewModel.hasChanged());
    }

    @Test
    public void setDragonsScore_null() {
        Game game = new Game();

        AdminGameViewModel adminGameViewModel = new AdminGameViewModel(game);
        adminGameViewModel.setDragonsScore(null);

        Assert.assertEquals("0", adminGameViewModel.getDragonsScore());
        Assert.assertTrue(adminGameViewModel.hasChanged());
    }

    @Test
    public void setDragonsScore_nullGameResult() {
        Game game = new Game();

        AdminGameViewModel adminGameViewModel = new AdminGameViewModel(game);
        adminGameViewModel.setDragonsScore("8");

        Assert.assertEquals("8", adminGameViewModel.getDragonsScore());
        Assert.assertTrue(adminGameViewModel.hasChanged());
    }

    @Test
    public void setDragonsScore_notNull() {
        Game game = new Game();
        game.gameResult = new GameResult();
        game.gameResult.dragonsScore = 99;

        AdminGameViewModel adminGameViewModel = new AdminGameViewModel(game);
        adminGameViewModel.setDragonsScore("3");

        Assert.assertEquals("3", adminGameViewModel.getDragonsScore());
        Assert.assertTrue(adminGameViewModel.hasChanged());
    }

    @Test
    public void getDragonsScore() {
        Game game = new Game();

        AdminGameViewModel adminGameViewModel = new AdminGameViewModel(game);

        Assert.assertEquals("", adminGameViewModel.getDragonsScore());
        Assert.assertFalse(adminGameViewModel.hasChanged());
    }
}

File: app/src/test/java/com/slapshotapps/dragonshockey/activities/admin/viewmodels/AdminGameViewModelTest.kt
package com.slapshotapps.dragonshockey.activities.admin.viewmodels

import com.slapshotapps.dragonshockey.models.Game
import com.slapshotapps.dragonshockey.models.GameResult
import junit.framework.Assert
import org.junit.Test

/**
 * Created on 10/23/16.
 */
class AdminGameViewModelTest {
    @Test
    fun getOpponentName_null() {
        val game = Game()

        val adminGameViewModel = AdminGameViewModel(game)

        Assert.assertNotNull(adminGameViewModel.opponentName)
    }

    @Test
    fun setOpponentName_null() {
        val game = Game()

        val adminGameViewModel = AdminGameViewModel(game)
        adminGameViewModel.opponentName = "test"

        Assert.assertEquals("test", adminGameViewModel.opponentName)

        Assert.assertTrue(adminGameViewModel.hasChanged())
    }

    @Test
    fun setOpponentName_notNull() {
        val game = Game()
        game.opponent = "beforeTest"

        val adminGameViewModel = AdminGameViewModel(game)
        adminGameViewModel.opponentName = "test"

        Assert.assertEquals("test", adminGameViewModel.opponentName)

        Assert.assertTrue(adminGameViewModel.hasChanged())
    }

    @Test
    fun setOpponentScore_null() {
        val game = Game()

        val adminGameViewModel = AdminGameViewModel(game)
        adminGameViewModel.opponentScore = null

        Assert.assertEquals("0", adminGameViewModel.opponentScore)
        Assert.assertTrue(adminGameViewModel.hasChanged())
    }

    @Test
    fun setOpponentScore_nullGameResult() {
        val game = Game()

        val adminGameViewModel = AdminGameViewModel(game)
        adminGameViewModel.opponentScore = "8"

        Assert.assertEquals("8", adminGameViewModel.opponentScore)
        Assert.assertTrue(adminGameViewModel.hasChanged())
    }

    @Test
    fun setOpponentScore_notNull() {
        val game = Game()
        game.gameResult = GameResult()
        game.gameResult!!.opponentScore = 2

        val adminGameViewModel = AdminGameViewModel(game)
        adminGameViewModel.opponentScore = "8"

        Assert.assertEquals("8", adminGameViewModel.opponentScore)
        Assert.assertTrue(adminGameViewModel.hasChanged())
    }

    @Test
    @Throws(Exception::class)
    fun getOpponentScore() {
        val game = Game()

        val adminGameViewModel = AdminGameViewModel(game)

        Assert.assertEquals("", adminGameViewModel.opponentScore)
        Assert.assertFalse(adminGameViewModel.hasChanged())
    }

    @Test
    fun setDragonsScore_null() {
        val game = Game()

        val adminGameViewModel = AdminGameViewModel(game)
        adminGameViewModel.dragonsScore = null

        Assert.assertEquals("0", adminGameViewModel.dragonsScore)
        Assert.assertTrue(adminGameViewModel.hasChanged())
    }

    @Test
    fun setDragonsScore_nullGameResult() {
        val game = Game()

        val adminGameViewModel = AdminGameViewModel(game)
        adminGameViewModel.dragonsScore = "8"

        Assert.assertEquals("8", adminGameViewModel.dragonsScore)
        Assert.assertTrue(adminGameViewModel.hasChanged())
    }

    @Test
    fun setDragonsScore_notNull() {
        val game = Game()
        game.gameResult = GameResult()
        game.gameResult!!.dragonsScore = 99

        val adminGameViewModel = AdminGameViewModel(game)
        adminGameViewModel.dragonsScore = "3"

        Assert.assertEquals("3", adminGameViewModel.dragonsScore)
        Assert.assertTrue(adminGameViewModel.hasChanged())
    }

    @Test
    fun getDragonsScore() {
        val game = Game()

        val adminGameViewModel = AdminGameViewModel(game)

        Assert.assertEquals("", adminGameViewModel.dragonsScore)
        Assert.assertFalse(adminGameViewModel.hasChanged())
    }
}